axum-openapi/src/lib.rs

```diff
@@ -0,0 +1,291 @@
+#![allow(
+    clippy::field_reassign_with_default,
+    clippy::new_without_default,
+    clippy::type_complexity,
+    dead_code,
+    unreachable_code,
+    unused_imports,
+    unused_mut,
+    unused_variables
+)]
+
+use axum::{
+    body::{Body, BoxBody, Bytes, Full, HttpBody},
+    http::{header, HeaderValue, Request, Response, StatusCode},
+    response::IntoResponse,
+    routing::Route,
+    BoxError, Json, Router,
+};
+use openapiv3::{HeaderStyle, MediaType, OpenAPI, Operation, PathItem, ReferenceOr, Responses};
+use std::{borrow::Cow, convert::Infallible, future::Future};
+use tower_layer::Layer;
+use tower_service::Service;
+
+pub mod handler_method_routing;
+
+pub struct OpenApiRouter<B = Body> {
+    router: Router<B>,
+    schema: OpenAPI,
+}
+
+impl<B> OpenApiRouter<B>
+where
+    B: Send + 'static,
+{
+    pub fn new() -> Self {
+        Self::with_openapi(Default::default())
+    }
+
+    pub fn with_openapi(openapi: OpenAPI) -> Self {
+        Self {
+            router: Default::default(),
+            schema: openapi,
+        }
+    }
+
+    pub fn route<T>(mut self, path: &str, service: T) -> Self
+    where
+        T: Service<Request<B>, Response = Response<BoxBody>, Error = Infallible>
+            + ToPathItem
+            + Clone
+            + Send
+            + 'static,
+        T::Future: Send + 'static,
+    {
+        // TODO(david): correct path templating with `{key}` instead of `:key`
+        // TODO(david): does openapi have wildcards?
```

Comment:
I found this weird in the changelog too.. By "wildcards", you mean multi-segment variables / placeholders, i.e. `*foo` as opposed to `:foo`, right?